components/kyma-environment-broker/internal/event-hub/event_hub_step.go

```diff
@@ -0,0 +1,209 @@
+package event_hub
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/ptr"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventhub/mgmt/2017-04-01/eventhub"
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/event-hub/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+const (
+	authorizationRuleName = "RootManageSharedAccessKey"
+
+	kafkaPort = 9093
+
+	k8sSecretNamespace                = "knative-eventing"
+	componentNameKnativeEventing      = "knative-eventing"
+	componentNameKnativeEventingKafka = "knative-eventing-kafka"
+	kafkaProvider                     = "azure"
+)
+
+// ensure the interface is implemented
+var _ process.Step = (*ProvisionAzureEventHubStep)(nil)
+
+type ProvisionAzureEventHubStep struct {
+	operationManager    *process.OperationManager
+	hyperscalerProvider azure.HyperscalerProvider
+	accountProvider     hyperscaler.AccountProvider
+	context             context.Context
+}
+
+func NewProvisionAzureEventHubStep(os storage.Operations, hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) *ProvisionAzureEventHubStep {
+	return &ProvisionAzureEventHubStep{
+		operationManager:    process.NewOperationManager(os),
+		accountProvider:     accountProvider,
+		context:             ctx,
+		hyperscalerProvider: hyperscalerProvider,
+	}
+}
+
+func (p *ProvisionAzureEventHubStep) Name() string {
+	return "Provision Azure Event Hubs"
+}
+
+func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperation,
+	log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+
+	hypType := hyperscaler.Azure
+
+	// parse provisioning parameters
+	pp, err := operation.GetProvisioningParameters()
+	if err != nil {
+		// if the parameters are incorrect, there is no reason to retry the operation
+		// a new request has to be issued by the user
+		log.Errorf("Aborting after failing to get valid operation provisioning parameters: %v", err)
+		return p.operationManager.OperationFailed(operation, "invalid operation provisioning parameters")
+	}
+	log.Infof("HAP lookup for credentials to provision cluster for global account ID %s on Hyperscaler %s", pp.ErsContext.GlobalAccountID, hypType)
+
+	// get hyperscaler credentials from HAP
+	credentials, err := p.accountProvider.GardenerCredentials(hypType, pp.ErsContext.GlobalAccountID)
```

Comment:
`GardenerCredentials` method underneath fetch secret or update it. Those operations could be temporary, so maybe `time.Duration` here should be returned? Also, it is good practice to not run steps forever (when you return `time.Duration`) so at the beginning of step `operation.UpdatedAt` field can be checked how long operation is in progress state.
Another question is, it is necessary step to make provisioning, if not step can return success with proper log error.

Reply:
Good catch. I didn't see the last call in the implementation. You are right, it makes sense to retry.

Concerning UpdatedAt, we implemented `retryOperation` which makes use of this field. Moving this to a common package could help others.